modin/experimental/engines/omnisci_on_native/test/utils.py

```diff
@@ -0,0 +1,216 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import modin.pandas as pd
+import pandas
+import pyarrow as pa
+
+from modin.pandas.test.utils import (
+    df_equals,
+    io_ops_bad_exc,
+    eval_io as general_eval_io,
+)
+
+from modin.experimental.engines.omnisci_on_native.frame.omnisci_worker import (
+    OmnisciServer,
+)
+
+
+def eval_io(
+    fn_name,
+    comparator=df_equals,
+    cast_to_str=False,
+    check_exception_type=True,
+    raising_exceptions=io_ops_bad_exc,
+    check_kwargs_callable=True,
+    modin_warning=None,
+    md_extra_kwargs=None,
+    *args,
+    **kwargs,
+):
+    """
+    Evaluate I/O operation and do equality check after importing Modin's data to OmniSci.
+
+    Notes
+    -----
+    For parameters description please refer to ``modin.pandas.test.utils.eval_io``.
+    """
+
+    def omnisci_comparator(df1, df2):
+        """Evaluate equality comparison of the passed frames after importing the Modin's one to OmniSci."""
+        with ForceOmnisciImport(df1, df2):
+            comparator(df1, df2)
+
+    general_eval_io(
+        fn_name,
+        comparator=omnisci_comparator,
+        cast_to_str=cast_to_str,
+        check_exception_type=check_exception_type,
+        raising_exceptions=raising_exceptions,
+        check_kwargs_callable=check_kwargs_callable,
+        modin_warning=modin_warning,
+        md_extra_kwargs=md_extra_kwargs,
+        *args,
+        **kwargs,
+    )
+
+
+class ForceOmnisciImport:
+    """
+    Trigger import execution for Modin DataFrames obtained by OmniSci engine if already not.
+
+    When using as a context class also cleans up imported tables at the end of the context.
+
+    Parameters
+    ----------
+    *dfs : iterable
+        DataFrames to trigger import.
+    """
+
+    def __init__(self, *dfs):
+        self._imported_frames = []
+        for df in dfs:
+            if not isinstance(df, (pd.DataFrame, pd.Series)):
+                continue
+            df.shape  # to trigger real execution
+            if df.empty:
+                continue
+            partition = df._query_compiler._modin_frame._partitions[0][0]
+            if partition.frame_id is not None:
```

Comment:
Shouldn't this be like `if partition.frame_id is None:`?

Reply:
`frame_id` holds the table name that is imported to OmniSci, which corresponds to the `partition` DataFrame. If `frame_id` is None, then no import has been triggered yet, so we have to stay in this for loop and import the table, if `frame_id` is not None, then the table has been already imported and we have to skip this frame.

Reply:
I got, thanks!